migrations/20241222143712-add-resource-type-delete-availability-in-attachments.js

```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  async up(db) {
+    const attachmentsCollection = db.collection('attachments')
+    await attachmentsCollection.updateMany({ availability: { $exists: true } }, { $unset: { availability: '' } })
+    await attachmentsCollection.updateMany(
+      { resourceType: { $exists: false } },
+      { $set: { resourceType: 'attachment' } }
+    )
+
+    await attachmentsCollection.updateMany({ resourceType: 'attachments' }, { $set: { resourceType: 'attachment' } })
+  },
+
+  async down(db) {
+    const attachmentsCollection = db.collection('attachments')
+    await attachmentsCollection.updateMany(
+      { availability: { $exists: false } },
+      {
+        $set: {
+          availability: {
+            status: 'open',
+            date: null
+          }
+        }
+      }
+    )
+    await attachmentsCollection.updateMany({ resourceType: { $exists: true } }, { $unset: { resourceType: '' } })
```

Comment:
I believe it would be better to replace the removal with reverting to resourceType: "attachments".
For documents without a resourceType field, that field could always be ignored. However, for fields with attachments, the field is currently deleted instead of being reverted to attachments. 
This could impact older application logic if someone needs to use an older version of the application and database by rolling back the migration.

Reply:
But where there is a resourceType field in the attachments document, it will be “attachments” anyway, so maybe it's better not to change this value at all when rolling back the migration?